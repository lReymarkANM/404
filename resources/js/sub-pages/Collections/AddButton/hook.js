import Cookies from 'js-cookie';
import useCollectionsList from '../CollectionsList/hook';

const useAddButton = () => {
  const { getCollections } = useCollectionsList();

  const addCollection = async () => {
    const response = await fetch('/api/collections/add', {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
        Authorization: `Bearer ${Cookies.get('token')}`,
        'X-CSRF-TOKEN': document.querySelector('meta[name=csrf_token]').content
      },
      body: JSON.stringify({ title: 'Untitled' })
    });
    if (!response.ok) return;
    getCollections();
  }

  return { addCollection };
}

export default useAddButton;