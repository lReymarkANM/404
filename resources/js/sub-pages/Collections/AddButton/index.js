import { FaPlus } from 'react-icons/fa';
import useAddButton from './hook';

const AddButton = () => {
  const { addCollection } = useAddButton();

  return (
    <div className="flex justify-center pb-2">
      <button
        className="flex justify-center items-center text-[25px] h-[40px] w-[40px] text-dark-gray rounded-md border border-dark-gray"
        onClick={() => addCollection()}
      >
        <FaPlus />
      </button>
    </div>
  );
}

export default AddButton;