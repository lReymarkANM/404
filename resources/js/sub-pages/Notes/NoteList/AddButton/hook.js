import { useSelector } from 'react-redux';
import Cookies from 'js-cookie';
import useNotes from '@sub-pages/Notes/hook';

const useAddButton = () => {
  const notes = useSelector(state => state.notes);
  const collectionId = notes.collectionId || Cookies.get('collectionId');
  const { getNotes } = useNotes(collectionId);
  
  const addNote = async () => {
    const response = await fetch(`/api/notes/add`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${Cookies.get('token')}`,
        'X-CSRF-TOKEN': document.querySelector('meta[name=csrf_token]').content
      },
      body: JSON.stringify({ collectionId })
    });
    if (!response.ok) return;
    getNotes();
  }

  return { addNote };
}

export default useAddButton;