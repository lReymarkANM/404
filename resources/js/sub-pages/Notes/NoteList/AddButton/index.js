import { FaPlus } from 'react-icons/fa';
import useAddButton from './hook';

const AddButton = () => {
  const { addNote } = useAddButton();

  return (
    <div className="mr-2">
      <button
        className="w-[40px] h-[40px] flex justify-center items-center border border-dark-gray text-[25px] text-dark-gray rounded-md"
        onClick={addNote}
      >
        <FaPlus />
      </button>
    </div>
  );
}

export default AddButton;