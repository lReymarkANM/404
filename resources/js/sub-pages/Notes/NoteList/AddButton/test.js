import 'regenerator-runtime';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import AddButton from './index';

test('NoteList Add Button renders correctly', () => {
  const initialState = { notes: [] };
  const mockStore = configureStore();

  const component = renderer.create(
    <Provider store={mockStore(initialState)}>
      <AddButton/>
    </Provider>
  );
  const NoteListAddButton = component.toJSON();
  expect(NoteListAddButton).toMatchSnapshot();
});