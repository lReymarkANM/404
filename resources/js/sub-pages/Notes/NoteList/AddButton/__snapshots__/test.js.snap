// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteList Add Button renders correctly 1`] = `
<div
  className="py-4 border-b border-dark-gray mb-4"
>
  <button
    className="w-[40px] h-[40px] flex justify-center items-center border border-dark-gray text-[25px] text-dark-gray rounded-md"
    onClick={[Function]}
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
      />
    </svg>
  </button>
</div>
`;
